server/src/main/java/io/crate/expression/scalar/ArrayUpperFunction.java

```diff
@@ -197,7 +198,7 @@ public Query toQuery(Function parent, Function arrayLength, Context context) {
                 if (cmpVal == 0) {
                     return IsNullPredicate.refExistsQuery(arrayRef, context);
                 }
-                return genericAndDocValueCount(parent, context, arrayRef, valueCountIsMatch);
+                return docValueCountOrGeneric(parent, context, arrayRef, valueCountIsMatch);
```

Comment:
Did you compare the performance of `docValueCountOrGeneric` and `genericAndDocValueCount`. If `genericAndDocValueCount` is faster we could consider to keep using it for numeric types.

Although I suspect the new variant might be faster anyways

Reply:
Before I saw the issue ^ I did run a benchmark and results not always good for should/should:
```
# Results (server side duration in ms)
V1: 5.0.0-20d203cf865ae2adc0e565a7410eafb5f01180af
V2: 5.0.0-176e118d0cbfe13e6ca2a6ad1a130a015e49e364

Q: select count(*) from t_arrays_biased_empty where array_length(xs, 1) > 1
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |       87.293 ±  522.279 |     11.365 |     12.427 |     13.915 |   6129.266 |
|   V2    |      222.661 ±  881.546 |    125.916 |    127.219 |    129.227 |  12143.896 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               +  87.35%                           + 164.40%   
There is a 93.76% probability that the observed difference is not random, and the best estimate of that difference is 87.35%
The test has no statistical significance

Q: select count(*) from t_arrays_biased_one_element where array_length(xs, 1) > 1
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |       85.769 ±  196.021 |      8.941 |     77.813 |     79.546 |   2583.126 |
|   V2    |      409.911 ±  439.778 |    200.190 |    278.196 |    456.460 |   3100.396 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               + 130.79%                           + 112.57%   
There is a 100.00% probability that the observed difference is not random, and the best estimate of that difference is 130.79%
The test has statistical significance

Q: select count(*) from t_arrays_biased_5_elements where array_length(xs, 1) > 1
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |     1845.616 ± 1162.076 |    464.447 |   2343.188 |   2561.893 |   6243.268 |
|   V2    |     2752.334 ± 1761.113 |    246.354 |   2214.474 |   4320.745 |   7403.481 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               +  39.44%                           -   5.65%   
There is a 100.00% probability that the observed difference is not random, and the best estimate of that difference is 39.44%
The test has statistical significance

Q: select count(*) from t_arrays_even where array_length(xs, 1) > 1
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      535.434 ±  264.616 |     63.493 |    573.027 |    602.357 |   2995.230 |
|   V2    |      287.111 ±  560.662 |    179.978 |    183.562 |    186.717 |   3716.548 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               -  60.38%                           - 102.95%   
```

it can be explained that first MUST was a short-circuit for some cases when we really had to break early - but updates test proofs that we cannot use `genericAndDocValueCount` anyway